sw/device/lib/testing/test_framework/check.h

```diff
@@ -209,4 +209,27 @@
     }                                                                \
   } while (false)
 
+/**
+ * Checks that the `status_t` represents a non-error value.
+ *
+ * Prints a human-readable error message if the status represents an error.
+ *
+ * @param expr An expression which evaluates to a `status_t`.
+ */
+#define CHECK_STATUS(expr, ...)                                               \
+  do {                                                                        \
+    status_t status_ = expr;                                                  \
```

Comment:
This header should include `sw/device/lib/base/status.h`, and the `//sw/device/lib/testing/test_framework:check` target should now depend on `//sw/device/lib/base:status`.